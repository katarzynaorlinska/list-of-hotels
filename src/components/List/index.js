/* eslint-disable array-callback-return */
import Item from "../Item";
import useSearch from "../../hooks/useSearch";

function List(props) {
  const { setSearch, searchResults } = useSearch(props.items);

  return (
    <div>
      <input
        type="text"
        placeholder="Search hotel.."
        onChange={(event) => {
          setSearch(event.target.value);
        }}
      ></input>
      <ul>
        {searchResults.map((item) => (
          <Item
            key={`hotel-element-main-list-${item.id}`}
            id={item.id}
            name={item.name}
            city={item.location.city}
            images={item.images}
            stars={item.stars}
          />
        ))}
      </ul>
    </div>
  );
}

export default List;
