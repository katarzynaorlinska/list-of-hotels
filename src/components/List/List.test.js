import { create } from "react-test-renderer";
import mockResponse from "../../mock/listHotel.json";
import List from "./index";

jest.mock("../Item", () => () => "Item");
jest.mock("../../hooks/useSearch", () => () => ({
  setSearch: () => jest.fn(),
  searchResults: mockResponse.hotels,
}));

test("renders correctly", () => {
  const tree = create(<List items={mockResponse.hotels} />).toJSON();
  expect(tree).toMatchSnapshot();
});
