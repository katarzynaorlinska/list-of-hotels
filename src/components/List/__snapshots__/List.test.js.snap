// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Search hotel.."
    type="text"
  />
  <ul>
    Item
    Item
    Item
    Item
    Item
    Item
    Item
    Item
    Item
    Item
  </ul>
</div>
`;
